### PR TITLE
Update Microsoft.Azure.Cosmos to 3.3.1

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

**Comment by dougbu** [comment]
@bricelam only questions: Why is the build failing? And, is this approved for preview2 (I suspect the change is ask-mode.)

**Comment by bricelam** [comment]
Looks like they have some xplat issues. Fails on Linux and macOS:
```
System.FormatException: The format of value '17:02:44 UTC 2019/X64 3.3.0/3.3.0-.NET Core 3.1.0-preview1.19480.27 64623 Microsoft.EntityFrameworkCore.Cosmos/3.1.0-ci' is invalid.
System.FormatException: The format of value '17.7.0: Sun Jun  2 20:31:42 PDT 2019; root:xnu-4570.71.46~1/RELEASE_X86_64/X64 3.3.0/3.3.0-.NET Core 3.1.0-preview1.19480.27 50250 Microsoft.EntityFrameworkCore.Cosmos/3.1.0-ci' is invalid.
   at System.Net.Http.Headers.HttpHeaderParser.ParseValue(String value, Object storeValue, Int32& index)
   at System.Net.Http.Headers.HttpHeaders.ParseAndAddValue(HeaderDescriptor descriptor, HeaderStoreItemInfo info, String value)
   at System.Net.Http.Headers.HttpHeaders.Add(String name, String value)
   at Microsoft.Azure.Documents.HttpClientExtension.AddUserAgentHeader(HttpClient httpClient, UserAgentContainer userAgent)
   at Microsoft.Azure.Cosmos.DocumentClient.Initialize(Uri serviceEndpoint, ConnectionPolicy connectionPolicy, Nullable`1 desiredConsistencyLevel, HttpMessageHandler handler, ISessionContainer sessionContainer, Nullable`1 enableCpuMonitor, IStoreClientFactory storeClientFactory)
   at Microsoft.Azure.Cosmos.DocumentClient..ctor(Uri serviceEndpoint, String authKeyOrResourceToken, EventHandler`1 sendingRequestEventArgs, ConnectionPolicy connectionPolicy, Nullable`1 desiredConsistencyLevel, JsonSerializerSettings serializerSettings, ApiType apitype, EventHandler`1 receivedResponseEventArgs, HttpMessageHandler handler, ISessionContainer sessionContainer, Nullable`1 enableCpuMonitor, Func`2 transportClientHandlerFactory, IStoreClientFactory storeClientFactory)
   at Microsoft.Azure.Cosmos.CosmosClient..ctor(String accountEndpoint, String authKeyOrResourceToken, CosmosClientOptions clientOptions)
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.SingletonCosmosClientWrapper.get_Client() in /_/src/EFCore.Cosmos/Storage/Internal/SingletonCosmosClientWrapper.cs:line 64
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.get_Client() in /_/src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs:line 91
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.CreateDatabaseIfNotExistsOnceAsync(DbContext _, Object __, CancellationToken cancellationToken) in /_/src/EFCore.Cosmos/Storage/Internal/CosmosClientWrapper.cs:line 136
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementationAsync[TState,TResult](Func`4 operation, Func`4 verifySucceeded, TState state, CancellationToken cancellationToken) in /_/src/EFCore/Storage/ExecutionStrategy.cs:line 268
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementationAsync[TState,TResult](Func`4 operation, Func`4 verifySucceeded, TState state, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.CleanAsync(DbContext context) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 159
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.Clean(DbContext context) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 154
   at Microsoft.EntityFrameworkCore.TestUtilities.TestStore.Initialize(Func`1 createContext, Action`1 seed, Action`1 clean) in /_/test/EFCore.Specification.Tests/TestUtilities/TestStore.cs:line 78
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.Initialize(Func`1 createContext, Action`1 seed, Action`1 clean) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 89
   at Microsoft.EntityFrameworkCore.TestUtilities.TestStore.Initialize(IServiceProvider serviceProvider, Func`1 createContext, Action`1 seed, Action`1 clean) in /_/test/EFCore.Specification.Tests/TestUtilities/TestStore.cs:line 48
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosTestStore.CreateInitialized(String name, Action`1 extensionConfiguration) in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosTestStore.cs:line 32
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosDbConfiguredConditionAttribute.TryConnectAsync() in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs:line 35
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosDbConfiguredConditionAttribute.TryConnectAsync() in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs:line 61
   at Microsoft.EntityFrameworkCore.Cosmos.TestUtilities.CosmosDbConfiguredConditionAttribute.IsMetAsync() in /_/test/EFCore.Cosmos.FunctionalTests/TestUtilities/CosmosDbConfiguredConditionAttribute.cs:line 24
   at Microsoft.EntityFrameworkCore.TestUtilities.Xunit.XunitTestCaseExtensions.TrySkipAsync(XunitTestCase testCase, IMessageBus messageBus) in /_/test/EFCore.Specification.Tests/TestUtilities/Xunit/XunitTestCaseExtensions.cs:line 43
   at Microsoft.EntityFrameworkCore.TestUtilities.Xunit.ConditionalTheoryTestCase.RunAsync(IMessageSink diagnosticMessageSink, IMessageBus messageBus, Object[] constructorArguments, ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource) in /_/test/EFCore.Specification.Tests/TestUtilities/Xunit/ConditionalTheoryTestCase.cs:line 34
```

**Comment by bricelam** [comment]
cc @j82w

File: eng/Versions.props [hunk 1]
```diff
@@ -16,7 +16,7 @@
   </PropertyGroup>
   <PropertyGroup Label="Dependencies from nuget.org">
     <CastleCorePackageVersion>4.4.0</CastleCorePackageVersion>
-    <MicrosoftAzureCosmosPackageVersion>3.2.0</MicrosoftAzureCosmosPackageVersion>
+    <MicrosoftAzureCosmosPackageVersion>3.3.0</MicrosoftAzureCosmosPackageVersion>
```
Comment (j82w) [inline eng/Versions.props]: @bricelam please upgrade to [3.3.1](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.3.1) which has the hotfix for the linux and Mac OS. It was released on Friday.

**Pull request merged**